asf_heat_pump_suitability/analysis/protected_areas/20250207_missing_protected_areas.py

```diff
@@ -0,0 +1,319 @@
+# %% [markdown]
+# ## Building conservation area data
+#
+# We have a dataset of building conservation areas from Historic England and Welsh Gov, covering Local Authority Districts (LADs) in England and Wales. Some LADs are missing building conservation area data. We want to assess whether we can make the assumption that if an LAD has ANY building conservation area data, then it can be assumed to be complete. We will test this assumption by looking at a few example areas.
+
+# %%
+import polars as pl
+import geopandas as gpd
+import pandas as pd
+import matplotlib.pyplot as plt
+from asf_heat_pump_suitability.getters import get_datasets
+from asf_heat_pump_suitability.pipeline.prepare_features import (
+    epc,
+    protected_areas,
+    lat_lon,
+)
+
+# %% [markdown]
+# ### Load and transform datasets
+
+# %%
+# Load England and Wales building conservation area data and concatenate
+e_gdf = (
+    get_datasets.load_gdf_historic_england_conservation_areas(
+        columns=["name", "geometry"]
+    )
+    .to_crs("EPSG:27700")
+    .rename(columns={"name": "sitename"})
+)
+w_gdf = get_datasets.load_gdf_welsh_gov_conservation_areas(
+    columns=["sitename", "geometry"]
+)
+full_cons_areas_gdf = pd.concat([e_gdf, w_gdf]).drop_duplicates(subset=["geometry"])
+full_cons_areas_gdf["in_conservation_area_ew"] = True
+
+# %%
+# Load geospatial boundaries of local authorities
+council_bounds = get_datasets.load_gdf_ons_council_bounds()
+
+# %%
+full_cons_areas_gdf["cons_area_size_m2"] = full_cons_areas_gdf["geometry"].area
+full_cons_areas_gdf = gpd.overlay(
```

Comment:
Also, just noticed that because we are doing the intersection, we are dropping over 400 rows from the original data.  A lot of this in places like Angelsey where I can see Beaumaris and the Menai Bridge, but also some villages in England as I spot check. I guess not to much to do since we are validating against the boundaries we have, but interesting as it could be another way we are losing data relative to boundaries that we do have.